src/api/realmGroups.ts

```diff
@@ -4,7 +4,22 @@ import { APIContext } from './context'
 
 type CreateRealmGroupData = {
   realmName: string
-  group: { name: string }
+  group: {
+    name: string
+    attributes: object
+  }
+}
+
+// transformAttributesForApi converts attributes to expected api format
+// { key1: 'value1' } becomes { key1: ['value1']}
+function transformAttributesForApi(payload: any) {
+  const attributes: any = {}
+  for (const [key, value] of Object.entries(payload.attributes || {})) {
+    attributes[key] = [value]
+  }
+  payload.attributes = attributes
+
+  return payload
```

Comment:
I am basically by-passing the typing by using `any`. The `attributes` day is dynamic in nature so hopefully this is ok.

I had

```
type ApiAttribute = { [prop: string]: [any] }

function transformAttributesForApi(payload: any) {
  const attributes: ApiAttribute = {}
  ...
```

but it seemed like overkill to me?

Reply:
i'm not familiar with group attrs. are they always a map of strings to an array of strings? or can the values be something else?

i'd probably use `Record<string, string>` or `Record<string, string[]>` despite the overkill

though there are enough `any`s floating around that it doesnt' make too much a difference 😆  i'd more prefer the explicit typing for documenting to future me what `attributes` are.

Reply:
On the server it is `key:string, value:string[]` but in the client we want the users to just use `key: string, value:string`
Keycloak uses an array of string values for some reason, even though their UI just supports a single value.

I'll give `Record<string, string>` a go, thanks.